terraform/validator/main.tf

```diff
@@ -0,0 +1,65 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+locals {
+  validator_config        = "parameters.yml"
+  final_validator_config  = "final_parameters.yml"
+  cloudwatch_agent_config = "agent_config.json"
+}
+
+resource "local_file" "update-validation-config" {
+  content = replace(replace(replace(replace(file("${var.test_dir}/${local.validator_config}"),
+    "<values_per_minute>", var.values_per_minute),
+    "<commit_hash>", var.cwa_github_sha),
+    "<commit_date>", var.cwa_github_sha_date),
+  "<cloudwatch_agent_config>", "${var.temp_directory}/${local.cloudwatch_agent_config}")
+
+  filename = "${var.test_dir}/${local.final_validator_config}"
+}
+
+resource "null_resource" "build-validator" {
+  provisioner "local-exec" {
+    command     = "make validator-build"
+    working_dir = split("test", var.test_dir)[0]
+  }
+
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+}
+
+// Uploading the validator to spending less time in 
+// and avoid memory issue in allocating memory with Windows
```

Comment:
Issues building go programs on windows seems like a problem we should root cause instead of working around.
We should be able to build _any_ go program on the test instance.

Reply:
I'm fine with root cause it. However, it would be better to `build it` since in the end, we would need to build it in order to build a `docker image` of the validator (e.g build a validator in docker image to be able to run it across platform, build a validator in docker image to be able to run on container environment - EKS, ECS). So I think it is not a work around but in the end it would be a permanent solutions if we want the validator to work in all platform.